### PR TITLE
Use public AWS addresses instead of private ones and add some minor fixes to aws

### PR DESCRIPTION
**Pull request by Tiboris**
Using private addresses will result into unreachable host
without vpn tunnel or mechanism to access AWS network.

Add security group, tags, keypair to init of AWS provider.
They have to be defined in provisioning config.

Remove asserts and replace them with ProvisioningErrors

Signed-off-by: Tibor Dudlák <tdudlak@redhat.com>

### CODE REVIEW

File: src/mrack/providers/aws.py [hunk 1]
```diff
@@ -101,20 +98,20 @@ async def create_server(self, req):
...
             InstanceType=specs.get("flavor"),
+            KeyName=self.ssh_key,
+            SecurityGroupIds=[self.sec_group],
         )
 
         ids = [srv.id for srv in aws_res]
-        assert len(ids) == 1  # ids must be len of 1 as we provision one vm at the time
-
+        if len(ids) != 1:  # ids must be len of 1 as we provision one vm at the time
+            raise (ProvisioningError)
```
Comment (netoarmando) [inline src/mrack/providers/aws.py:107]:
```python
raise ProvisioningError("Unexpected number of instances provisioned.")
```
_That is just a suggestion, feel free to improve the message._

File: src/mrack/providers/aws.py [hunk 2]
```diff
@@ -146,11 +140,11 @@ async def wait_till_provisioned(self, aws_id):
...
-        assert len(response["Reservations"]) == 1
-        assert len(response["Reservations"][0]["Instances"]) == 1
-        result = response["Reservations"][0]["Instances"][0]
-        # returns dict with aws instance information
-        return result
+
+        try:  # returns dict with aws instance information
+            return response["Reservations"][0]["Instances"][0]
+        except (KeyError, IndexError):
+            raise (ProvisioningError)
```
Comment (netoarmando) [inline src/mrack/providers/aws.py:147]:
```python
raise ProvisioningError("Missing data provisioned instance.")
```
_That is just a suggestion, feel free to improve the message._

**Review by netoarmando — COMMENTED** [review]
Nice improvements! Thanks for that. Please, consider the suggestions I left when raising exceptions.

File: src/mrack/providers/aws.py (hunk 2 above)
Comment (Tiboris) [reply src/mrack/providers/aws.py:147]: Will do :)

File: src/mrack/providers/aws.py (hunk 1 above)
Comment (Tiboris) [reply src/mrack/providers/aws.py:107]: Sure!

**Review by netoarmando — APPROVED** [review]

**Pull request merged**